mllib/src/main/scala/org/apache/spark/ml/source/libsvm/LibSVMRelation.scala

```diff
@@ -91,12 +91,10 @@ private[libsvm] class LibSVMFileFormat extends TextBasedFileFormat with DataSour
     val numFeatures: Int = libSVMOptions.numFeatures.getOrElse {
       // Infers number of features if the user doesn't specify (a valid) one.
       val dataFiles = files.filterNot(_.getPath.getName startsWith "_")
-      val path = if (dataFiles.length == 1) {
-        dataFiles.head.getPath.toUri.toString
-      } else if (dataFiles.isEmpty) {
+      val path = if (dataFiles.isEmpty) {
         throw new IOException("No input path specified for libsvm data")
       } else {
-        throw new IOException("Multiple input paths are not supported for libsvm data.")
+        dataFiles.map(_.getPath.toUri.toString).mkString(",")
```

Comment:
Sorry for interrupting but I believe I am working on an issue around this code path - https://issues.apache.org/jira/browse/SPARK-21326

Could we allow multiple paths and print warning maybe? I think it'd be similar ones with JSON/CSV schema inferences. These are also, up to my knowledge, discouraged in production but I we allow it with full scan.